File: src/main/java/examples/GreetingActor_2.java
package examples;

import akka.actor.AbstractActor;
import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;

public class GreetingActor_2 extends AbstractActor {

    public static void main(String[] args) throws InterruptedException {
        ActorSystem system = ActorSystem.create("MySystem");

        ActorRef greetingActor = system.actorOf(Props.create(GreetingActor_2.class), "greeter");
        greetingActor.tell(new SayHello("Pope Benedict"), ActorRef.noSender());

        // There are better ways to ensure message are received before termination
        Thread.sleep(100);
        system.terminate();
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
            .match(SayHello.class, hello -> {
                System.out.println("Hello " + hello.name);
                sender().tell(hello.name, self());
            }).build();
    }

    static class SayHello {
        public final String name;

        public SayHello(String n) {
            name = n;
        }
    }
}

File: src/main/kotlin/examples/GreetingActor_2.kt
package examples

import akka.actor.ActorRef
import akka.actor.ActorSystem
import akka.actor.Props
import akka.actor.UntypedActor

class SayHello(val name: String)

class GreetingActor_2 : UntypedActor() {

    override fun onReceive(msg: Any?) = when (msg) {
        is SayHello -> {
            println("Hello " + msg.name)
            sender().tell(msg.name, self())
        }
        else -> {
            unhandled(msg)
        }
    }
}

fun main(args: Array<String>) {
    val system = ActorSystem.create("MySystem")

    val greetingActor = system.actorOf(Props.create(GreetingActor_2::class.java), "greeter")
    greetingActor.tell(SayHello("Pope Benedict"), ActorRef.noSender())

    // There are better ways to ensure message are received before termination
    Thread.sleep(100)
    system.terminate()
}